snuba/query/parser/expressions.py

```diff
@@ -1,28 +1,181 @@
-from typing import Any, Optional
+from dataclasses import replace
+from typing import List, Tuple, Union
 
-from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from parsimonious.grammar import Grammar  # type: ignore
+from parsimonious.nodes import Node, NodeVisitor  # type: ignore
+from typing import Any, Iterable, Optional
+
+from snuba.query.expressions import (
+    Column,
+    CurriedFunctionCall,
+    Expression,
+    FunctionCall,
+    Literal,
+)
 from snuba.query.parser.functions import parse_function_to_expr
 from snuba.util import is_function, QUOTED_LITERAL_RE
 
+minimal_clickhouse_grammar = Grammar(
```

Comment:
If this seems over engineered, it is because it probably is. But please look at the unit test. All those use cases are actually happening in prod, so I don't think we have a lot of alternatives to actually parsing clickhouse expressions.

Reply:
It might be helpful to flag these queries somehow so that we can quickly find them in the query log (when implemented) and fix them at the source, and then deprecate this parsing behavior.

Reply:
:+:1 We can add snuba flavour functions via an extension that offers the same result and update the call sites in sentry.